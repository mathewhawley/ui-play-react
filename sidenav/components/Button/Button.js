import React, { PropTypes } from 'react';
import classnames from 'classnames';
import styles from './Button.scss';

const Button = ({
  onClick,
  icon,
  customStyles,
}) => {
  const classes = classnames(styles.base, customStyles, 'material-icons');
  return (
    <button className={classes} onClick={onClick}>{icon}</button>
  );
};

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  icon: PropTypes.string.isRequired,
  customStyles: PropTypes.string,
};

export default Button;
