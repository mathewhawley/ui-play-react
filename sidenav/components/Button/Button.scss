.base {
  min-width: 48px;
  min-height: 48px;
}
