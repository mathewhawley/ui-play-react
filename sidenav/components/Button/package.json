{
  "private": true,
  "name": "Button",
  "main": "./Button.js"
}
