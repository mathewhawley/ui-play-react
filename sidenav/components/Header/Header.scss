@import '../../styles/variables';

.base {
  background-color: $dark-grey;
  height: 60px;
  padding: 0 $global-padding;
  display: flex;
  align-items: center;
}

.button {
  color: $white;
  margin-left: -#{$global-padding - 10px};
}
