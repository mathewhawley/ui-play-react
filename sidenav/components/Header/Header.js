import React, { PropTypes } from 'react';
import Button from '../Button';
import styles from './Header.scss';

const Header = ({ toggleSideNav }) => (
  <div className={styles.base}>
    <Button
      icon='menu'
      onClick={toggleSideNav}
      customStyles={styles.button}
    />
  </div>
);

Header.propTypes = {
  toggleSideNav: PropTypes.func.isRequired,
};

export default Header;
