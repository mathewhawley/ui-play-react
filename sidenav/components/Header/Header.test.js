/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Header from './Header';
import Button from '../Button';

describe('<Header />', () => {
  let wrapper;
  let clickHandler;

  beforeEach(() => {
    clickHandler = jest.fn();
    wrapper = shallow(<Header toggleNav={clickHandler} />);
  });

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('renders correctly', () => {
    const tree = renderer.create(
      <Header toggleNav={clickHandler} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call click handler on click', () => {
    wrapper.find(Button).simulate('click');
    expect(clickHandler).toHaveBeenCalled();
  });
});
