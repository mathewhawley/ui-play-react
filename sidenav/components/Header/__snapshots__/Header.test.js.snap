exports[`<Header /> renders correctly 1`] = `
<div
  className="base">
  <button
    className="base button material-icons"
    onClick={[Function]}>
    menu
  </button>
</div>
`;
